### PR TITLE
Update Jasmine to 3.0

### PR DESCRIPTION
**Pull request by Roaders**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by Roaders** [comment]
Documentation for new matchers:
https://jasmine.github.io/api/3.3/matchers.html

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -453,6 +454,11 @@ declare namespace jasmine {
         toThrow(expected?: any): boolean;
         toThrowError(message?: string | RegExp): boolean;
         toThrowError(expected?: new (...args: any[]) => Error, message?: string | RegExp): boolean;
+        toThrowMatching(predicate: (thrown: any) => boolean): boolean;
+        nothing(expectationFailOutput?: any): boolean;
```
Comment (pe8ter) [inline types/jasmine/index.d.ts]: The `nothing` matcher already exists in the `NothingMatcher` interface.

**Comment by pe8ter** [comment]
I added one comment to this PR that should be addressed before merging. Aside from that issue, I think this PR should go in before mine (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30366).

**Comment by Roaders** [comment]
> I added one comment to this PR that should be addressed before merging. Aside from that issue, I think this PR should go in before mine (#30366).

I've now removed `nothing` from `Matcher<T>`

**Review by pe8ter — APPROVED** [review]

**Review by lukas-zech-software — APPROVED** [review]
lgtm 👍

**Review by pe8ter — APPROVED** [review]

**Comment by PranavSenthilnathan** [comment]
The version should only be MAJOR.MINOR (no revision number). This is a rule that should be caught by the dt-header lint rule, so you should turn it on and fix it. I'll go ahead and approve this PR and @pe8ter can update it in his PR.

**Pull request merged**